int/tests/test_proxy_mediator_connections.py

```diff
@@ -0,0 +1,98 @@
+from functools import partial
+from os import getenv
+
+from acapy_client import Client
+from acapy_client.api.connection import (
+    create_invitation,
+    get_connection,
+    receive_invitation,
+)
+from acapy_client.models.conn_record import ConnRecord
+from acapy_client.models.create_invitation_request import CreateInvitationRequest
+from acapy_client.models.receive_invitation_request import ReceiveInvitationRequest
+from acapy_client.types import Unset
+import pytest
+
+from . import record_state
+
+
+@pytest.fixture(scope="session")
+def agent_alice():
+    AGENT_ALICE = getenv("AGENT_ALICE", "http://agent_alice:4011")
+    return Client(base_url=AGENT_ALICE)
+
+
+@pytest.fixture(scope="session")
+def agent_bob():
+    AGENT_BOB = getenv("AGENT_BOB", "http://agent_bob:4012")
+    return Client(base_url=AGENT_BOB)
+
+
+@pytest.fixture
+async def agent_fixture():
+    def _agent_fixture(agent):
+        return agent
+
+    yield _agent_fixture
+
+
+@pytest.fixture
+async def create_connection(agent_fixture):
+    """Factory fixture to create a connection with
+    sender and receiver as parameters"""
+
+    async def _create_connection(sender: Client, receiver: Client):
+        sender = agent_fixture(sender)
+        receiver = agent_fixture(receiver)
+        invite = await create_invitation.asyncio(
+            client=sender,
+            json_body=CreateInvitationRequest(),
+            auto_accept=True,
+        )
+        assert invite
+        assert not isinstance(invite.invitation, Unset)
+        connection = await receive_invitation.asyncio(
+            client=receiver,
+            json_body=ReceiveInvitationRequest.from_dict(invite.invitation.to_dict()),
+            auto_accept=True,
+        )
+        return (invite, connection)
+
+    yield _create_connection
+
+
+agents = [
+    ("agent_alice", "agent_bob", "http://agent_alice:4011"),
+    ("agent_bob", "agent_alice", "http://reverse-proxy"),
+]
+
+
+@pytest.mark.parametrize("sender, receiver, endpoint", agents)
+@pytest.mark.asyncio
+async def test_connection_from_alice(
+    sender, receiver, endpoint, create_connection, request
+):
+    sender = request.getfixturevalue(sender)
+    receiver = request.getfixturevalue(receiver)
+    invite, connection = await create_connection(sender, receiver)
+    assert invite.invitation.service_endpoint == endpoint
+
+    invitation_sender = await get_connection.asyncio(
+        conn_id=invite.connection_id, client=sender
+    )
+    assert invitation_sender
+    connection_receiver = await get_connection.asyncio(
+        conn_id=connection.connection_id, client=receiver
+    )
+    assert connection_receiver
```

Comment:
These values don't seem to be used anywhere. What's our goal with these calls?

Reply:
I'd created these calls to retrieve the `state`. No longer necessary with the implementation of the `record_state()` function